### DISCUSSION TITLE
Adding mikro-orm dependency to a vite project causes the build process to fail.

### DISCUSSION THREAD

**Discussion by kyllerss** [mikro-orm/mikro-orm#3635] [Q&A]
**Describe the bug**
I have been using SvelteKit (which relies on Vite underneath) to develop an app that uses Mikro-orm. Development mode works well. However, when I attempt to run a build to prepare for a deployment, I am getting errors such as:

```
✘ [ERROR] Could not resolve "path"

    node_modules/picomatch/lib/constants.js:3:21:
      3 │ const path = require('path');
        ╵                      ~~~~~~

  The package "path" wasn't found on the file system but is built into node. Are you trying to
  bundle for node? You can use "platform: 'node'" to do that, which will remove this error.
```
I reviewed existing issues such as [this one](https://github.com/mikro-orm/mikro-orm/issues/3508) and [this other one](https://github.com/mikro-orm/mikro-orm/discussions/3068), but none of the suggestions seem to address the issue I am running into. 

I have created a paired-down reproduction [here](https://github.com/kyllerss/bug-vite-build-mikro). Any pointers would be greatly appreciated!

**Stack trace**
```
error during build:
Error: Build failed with 134 errors:
node_modules/@mikro-orm/core/EntityManager.js:4:23: ERROR: Could not resolve "util"
node_modules/@mikro-orm/core/MikroORM.js:143:40: ERROR: Could not resolve "@mikro-orm/seeder"
node_modules/@mikro-orm/core/connections/Connection.js:4:22: ERROR: Could not resolve "url"
node_modules/@mikro-orm/core/entity/ArrayCollection.js:4:23: ERROR: Could not resolve "util"
node_modules/@mikro-orm/core/entity/EntityAssigner.js:4:23: ERROR: Could not resolve "util"
...
    at failureErrorWithLog (/home/my-machine/projects/bugs/vite-build-mikro/node_modules/esbuild/lib/main.js:1566:15)
    at /home/my-machine/projects/bugs/vite-build-mikro/node_modules/esbuild/lib/main.js:1024:28
    at /home/my-machine/projects/bugs/vite-build-mikro/node_modules/esbuild/lib/main.js:969:67
    at buildResponseToResult (/home/my-machine/projects/bugs/vite-build-mikro/node_modules/esbuild/lib/main.js:1022:7)
    at /home/my-machine/projects/bugs/vite-build-mikro/node_modules/esbuild/lib/main.js:1134:14
    at responseCallbacks.<computed> (/home/my-machine/projects/bugs/vite-build-mikro/node_modules/esbuild/lib/main.js:671:9)
    at handleIncomingPacket (/home/my-machine/projects/bugs/vite-build-mikro/node_modules/esbuild/lib/main.js:726:9)
    at Socket.readFromStdout (/home/my-machine/projects/bugs/vite-build-mikro/node_modules/esbuild/lib/main.js:647:7)
    at Socket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:324:12)

```

**To Reproduce**
1. Clone reproduction project: https://github.com/kyllerss/bug-vite-build-mikro
2. run `npm run build`.

**Expected behavior**
The build to be completed successfully.

**Additional context**
The reproduction project is the templated demo SvelteKit project with the addition of `mikro-orm`. There is an *entity* thrown in there to reproduce the issue, but it isn't meant to be working code. The only other dependency of note is `netlify`.

**Versions**

| Dependency | Version |
| - | - |
| node | v18.7.0 |
| typescript | 4.8.4 |
| mikro-orm | 5.4.2 |
| your-driver | mikro-orm/postgresql |

**Comment by B4nan** [comment]
I'd say you have a higher chance of finding the solution if you ask vite community instead, I'll be happy to take a look, but it's quite a distant toolchain to me. The problem sounds a bit similar to what happens with webpack, so the information [here](https://mikro-orm.io/docs/deployment) might be somehow valid.

> You can use "platform: 'node'" to do that, which will remove this error.

Have you tried that? I don't see it in your vite config.
 
> You can mark the path "@mikro-orm/seeder" as external to exclude it from the bundle, which will
> remove this error. You can also surround this "require" call with a try/catch block to handle this
> failure at run-time instead of bundle-time.

You will need to do this too for sure.

**Reply by kyllerss** [reply]
No worries. Thank you for the reply. I'll reach out to the Vite community.
